integration_tests/steps/balancing_market.py

```diff
@@ -119,16 +119,14 @@ def follow_device_registry_energy_rate(context, device_name):
     positive_energy_rate = DeviceRegistry.REGISTRY[device_name][1]
     house1 = next(filter(lambda x: x.name == "House 1", context.simulation.area.children))
     house2 = next(filter(lambda x: x.name == "House 2", context.simulation.area.children))
-    iaa_fee = context.simulation.simulation_config.iaa_fee
-    add_val = [0, 2*iaa_fee/100 + iaa_fee/100**2, iaa_fee/100]
```

Comment:
Before moving on, I will take some time to dig a bit deeper on exactly how/why this was implemented. I will come back on the review with a response whether it really makes sense to remove this, or not.

Reply:
It was what I was afraid of. Apparently the changes break the operation of the balancing market if I understand correctly. I tried to run:
`d3a -l WARN run --setup balancing_market.default_2a  -t 15s -s 15m `
which is one of the setup files that the integration tests use. There were multiple cases where the following log was observed:
`
15:40:57.500 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {96eb4e} [BA House 2 -> House 2 Reserve] -0.005000000000000001 kWh (partial) @ 0.16831650000000006 -33.66330000000001 d4cec2a7-20ae-4fe6-a505-e46bbe3634bd
15:40:57.500 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {584ea4} [BA House 1 -> BA House 2] -0.005000000000000001 kWh (partial) @ 0.16500000000000004 -33.0 ab74a59f-3da9-4462-8f41-d80213cd0ffa
15:40:57.501 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {97c3e3} [H1 General Load -> BA House 1] -0.005000000000000001 kWh (partial) @ 0.16336633663366337 -32.67326732673267 ed3b34dd-e5d9-4010-b959-93935f259449
15:40:57.501 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {ea90d3} [BA House 2 -> House 2 Reserve] 0.005000000000000001 kWh (partial) @ 0.17851750000000002 35.7035 117b5d4d-6e46-4371-962e-4396dfb11715
15:40:57.502 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {a0c8fd} [BA House 1 -> BA House 2] 0.005000000000000001 kWh (partial) @ 0.17500000000000004 35.0 6dbb198c-e185-4f76-9501-fa5d2cec5f5f
15:40:57.502 WARNING  ( 148) d3a.models.market.balancing   : [BALANCING_TRADE][2019-03-20T16:00] {3668f7} [H1 Storage2 -> BA House 1] 0.005000000000000001 kWh (partial) @ 0.17326732673267328 34.65346534653465 a9b16405-ceb7-4f7f-a27c-a6d5ce77630a
`
The device registry is the following:
`
device_registry_dict = {
    "H1 General Load": (33, 35),
    "H2 General Load": (33, 35),
    "H1 Storage1": (33, 35),
    "H1 Storage2": (33, 35),
}
`
The evident problem is that the devices do not actually get paid according to the device registry, but less than that.  More specifically, Storage2 uses an energy rate of 34.65346534653465, while 35 was configured by the device registry. Note that this problem does not seem to affect the spot market, only the balancing market is affected by that. To summarise, I strongly suggest to revert the integration tests to their original form, since they obviously revealed the issue correctly, and to try to detect the reason why the balancing market in some cases does not use the device registry rate for the devices.

Reply:
Sorry, I dont get it, Faizan only added spaces here. This implementation was done by me in an earlier ticket.
Before that the test was done only in integer space which IMO is not sufficient. Lets discuss offline.